### PR TITLE
Alec dev2

### PR DESCRIPTION
**Pull request by AlecMcBurney**
Prayers

### CODE REVIEW

File: FileProcessor/src/main/java/elements/filters/CountFilter.java [hunk 1]
```diff
@@ -1,55 +1,139 @@
...
+    /**
+     * The process of the count filter.
+     * Checks to see if an entry is a file.
+     * Checks to see if the contents of the entry contains a string min number of times
+     * 
+     */
+    @Override
+    public void process() {        
+        // Make a local reference of the input of entries
+        ArrayList<Entry> input = this.getInputEntries();
```
Comment (Fluff-E) [inline FileProcessor/src/main/java/elements/filters/CountFilter.java:42]: Need setInputEntries method to work in sequencer.

File: FileProcessor/src/main/java/elements/filters/LengthFilter.java [hunk 2]
```diff
@@ -1,48 +1,108 @@
...
+     */
+    public LengthFilter(String operator, Long length) {
+        super();
+    }
+
+    @Override
+    public void process() {
+
+        // Make a local reference of the input of entries
+        ArrayList<Entry> input = this.getInputEntries();
```
Comment (Fluff-E) [inline FileProcessor/src/main/java/elements/filters/LengthFilter.java:37]: Need setInputEntries()

File: FileProcessor/src/main/java/elements/filters/NameFilter.java [hunk 3]
```diff
@@ -1,39 +1,127 @@
...
+     * @param key
+     */
+    public NameFilter(String key) {
+        super();
+    }
+
+    @Override
+    public void process() {
+        // Make a local reference of the input of entries
+        ArrayList<Entry> input = this.getInputEntries();
```
Comment (Fluff-E) [inline FileProcessor/src/main/java/elements/filters/NameFilter.java:33]: Need setInputEntries()

**Review by Fluff-E — CHANGES_REQUESTED** [review]
In order for each element to be automatically run it needs to have a setInputEntries() method for input, and a get.OutputEntries() for sending the output entries down the line.

File: FileProcessor/src/main/java/elements/filters/ContentFilter.java [hunk 4]
```diff
@@ -1,39 +1,102 @@
...
+            try {
+                // Checks if the key appears within the contents of the entry
+                if (contains(key, input.get(i).getContents())) {
+                    output.add(input.get(i));
+                }
+            } catch (FolderDoesNotContainTextException ex) {
+                System.out.println("Error in ContentFilter.java\n" + ex);
+            }
+        }
+
```
[the remark this answers is not in the document]
Comment (Tacitor) [reply FileProcessor/src/main/java/elements/filters/ContentFilter.java:47]: I think this is fine. To be best of my understanding the next assignment will be taken care of on the sequencer.

**Review by Tacitor — APPROVED** [review]
good

**Review by Fluff-E — APPROVED** [review]

**Pull request merged**